Wrappers/Python/ccpi/viewer/utils/io.py

```diff
@@ -43,3 +71,691 @@ def SaveRenderToPNG(render_window, filename):
+        # check origin and spacing are set
+        # check resampling and cropping are set
+        child_attributes_schema = Schema({
+            'resampled': bool,
+            'cropped': bool,
+            'spacing': Or(list, tuple),
+            'origin': Or(list, tuple),
+            Optional('resample_z'): bool,
+            Optional(str): object # allow any other keys and values
+        })
+        child_attributes_schema.validate(attributes)
+
+    def Write(self):
+        for var in [self._FileName, self._OriginalDatasetAttributes]:
+            if var is None:
+                raise Exception("file_name, dataset(/s) and attribute(/s), are required.")
+        for var in [self._ChildDatasets, self._ChildDatasetsAttributes]:
+            if var is []:
+                raise Exception("child dataset(/s) and attribute(/s), are required.")
+        
+        with h5py.File(self._FileName, 'w') as f:
+            
+            # give the file some important attributes
+            f.attrs['file_name'] = self._FileName
+            f.attrs['viewer_version'] = version
+            f.attrs['file_time'] = str(datetime.datetime.utcnow())
+            f.attrs['creator'] = np.string_('io.py')
+            f.attrs['HDF5_Version'] = h5py.version.hdf5_version
+            f.attrs['h5py_version'] = h5py.version.version
+            
+            # create the NXentry group
+            nxentry = f.create_group('entry1/tomo_entry')
+            nxentry.attrs['NX_class'] = 'NXentry'
+
+            datasets = [self._OriginalDataset]
+            datasets += self._ChildDatasets
+
+            attributes = [self._OriginalDatasetAttributes]
+            attributes+= self._ChildDatasetsAttributes
+
+            for i, dataset in enumerate(datasets):
+                data = dataset
+                dataset_info = attributes[i]
+                entry_num = i+1
+                dataset_name = 'entry{}/tomo_entry/data/data'.format(entry_num)
+
+                if data is not None:
+                    # The function imgdata.GetPointData().GetScalars() returns a pointer to a
+                    # vtk<TYPE>Array where the data is stored as X-Y-Z.
+                    array = numpy_support.vtk_to_numpy(
+                        data.GetPointData().GetScalars())
+
+                    # Note that we flip the shape here because
+                    # VTK's order is Fortran whereas h5py writes in
+                    # C order. We don't want to do deep copies so we write
+                    # with shape flipped and pretend the array is
+                    # C order.
+                    array = array.reshape(data.GetDimensions()[::-1])
+                else:
+                    array = None
+                try:
+                    if array is None:
+                        dset = f.create_dataset(dataset_name, dataset_info['shape'])
+                    else:
+                        dset = f.create_dataset(dataset_name, data=array)
```

Comment:
As an enhancement, I think it makes sense to use [chunking ](https://github.com/paskino/CIL/blob/3105be92c69fbc54c976e729030f85a1144c3fb6/Wrappers/Python/cil/io/NEXUSDataWriter.py#L139-L155)here.

An example of usage of that functionality can be found [here](https://github.com/paskino/CIL-work/blob/db4d4d227eacc86c9111992359c147492594db85/PSDI/test_chunking.py#L66-L74)

Reply:
Ok I will have a go!

Reply:
yes. It doesn't change speed if we read the whole dataset, but it could massively if we read only one slice at a time.